### PR TITLE
GPII-2531: Add tray icon

### PR DESCRIPTION
**Pull request by georgitodorov**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -276,34 +281,81 @@ fluid.onUncaughtException.addListener(function (err) {
...
+    gradeNames: "fluid.modelComponent",
     members: {
         tray: {
             expander: {
                 funcName: "gpii.app.makeTray",
-                args: ["{that}.options.icon"]
+                args: ["{that}.model.icon"]
             }
+        },
+        icons: {
```
Comment (michelled) [inline src/app.js]: Is there a reason to put this in 'members' rather than directly in the 'options' structure?

File: src/app.js [hunk 2]
```diff
@@ -276,34 +281,81 @@ fluid.onUncaughtException.addListener(function (err) {
...
     },
+    model: {
+        keyedInUserToken: null,
+        icon: "{that}.icons.keyedOut"
+    },
+    modelRelay: {
+        keyedInUserToken: {
+            target: "icon",
+            singleTransform: {
+                type: "fluid.transforms.free",
```
Comment (michelled) [inline src/app.js]: I think we could use a conditional transform here and get rid of the 'getTrayIcon' function. 
http://docs.fluidproject.org/infusion/development/ModelTransformationAPI.html#conditional-transform-fluid-transforms-condition-

File: src/app.js (hunk 1 above)
Comment (danailbd) [reply src/app.js]: Hmm, it really sounds reasonable for the icons to be configurable. We'll update it according to your advice :)

File: src/app.js (hunk 2 above)
Comment (danailbd) [reply src/app.js]: Yeah, it'd definitely make things prettier

File: src/app.js (hunk 1 above)
Comment (michelled) [reply src/app.js]: It's still configurable in the options block, isn't it? Am I missing something about how you'd like to configure it?

File: src/app.js [hunk 3]
```diff
@@ -274,36 +279,70 @@ fluid.onUncaughtException.addListener(function (err) {
...
         }
     },
+    model: {
+        keyedInUserToken: null,
+        icon: "{that}.options.icons.keyedOut"
+    },
+    modelRelay: {
+        keyedInUserToken: {
+            target: "icon",
+            singleTransform: {
```
Comment (danailbd) [inline src/app.js]: `fluid.transforms.condition` seems not to working properly with falsy values - it returns `undefined` if the condition is `null` (which is the case in the keyed out state). When the user is keyed in, the `true` icon is used; but after keyed out, the icon is left unchanged... An inner `free transform` for the condition should do the job, but it's not improvement on the initial version. @michelled, what do you do in such cases?

Comment (amb26) [reply src/app.js]: I recommend using the valueMapper transform which can map arbitrary input values including null and undefined. Sorry I can't link to our docs site which is inaccessible today but here is the relevant section of the source .md : https://github.com/fluid-project/infusion-docs/blob/master/src/documents/ModelTransformationAPI.md#mapping-based-on-input-value-fluidtransformsvaluemapper

@kaspermarkus, do you recall why we made this special treatment of null in the condition transform?

Comment (amb26) [reply src/app.js]: Here's that link again - http://docs.fluidproject.org/infusion/development/ModelTransformationAPI.html#mapping-based-on-input-value-fluid-transforms-valuemapper-

**Pull request merged**